staging/src/k8s.io/mount-utils/mount_linux_test.go

```diff
@@ -545,3 +549,220 @@ func mountArgsContainOption(t *testing.T, mountArgs []string, option string) boo
 
 	return strings.Contains(mountArgs[optionsIndex], option)
 }
+
+func TestIsMountPointUsingOpenAt2(t *testing.T) {
+	tests := map[string]struct {
+		expectedErr        error
+		expectedMountPoint bool
+		// returns the path to be tested for.
+		// returns an error if something went wrong during prepare.
+		prepare     func(*testing.T) (string, error)
+		openat2Mock func(int, string, *unix.OpenHow) (fd int, err error)
+	}{
+		"path is not absolute": {
+			expectedErr:        errPathNotAbsolute,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				return "../hello/", nil
+			},
+		},
+		"path is a symlink": {
+			expectedErr:        unix.ELOOP,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				tmpDir, err := os.MkdirTemp("", "symlink")
+				if err != nil {
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.RemoveAll(tmpDir)
+				})
+
+				f, err := os.CreateTemp(tmpDir, "some")
+				if err != nil {
+					return "", err
+				}
+
+				symlink := filepath.Join(tmpDir, "symlink")
+				err = os.Symlink(f.Name(), symlink)
+				if err != nil {
+					return "", err
+				}
+				return symlink, nil
+			},
+		},
+		"path is a mount point": {
+			expectedErr:        nil,
+			expectedMountPoint: true,
+			prepare: func(t *testing.T) (string, error) {
+				// /proc is a known mount-point and since mount-utils heavily relies on /proc/mounts
+				// it is a good option where we can test for mount points in non-privileged mode.
+				return "/proc", nil
+			},
+		},
+		"path is not a mount point": {
+			expectedErr:        nil,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				source, err := os.MkdirTemp("", "source")
+				if err != nil {
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.RemoveAll(source)
+				})
+				return source, nil
+			},
+		},
+		"path is a magiclink": {
+			expectedErr:        unix.ELOOP,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				// only /proc/1/exe is valid
+				return "/proc/1/exe", nil
+			},
+		},
+		"path does not exist": {
+			expectedErr:        os.ErrNotExist,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				f, err := os.CreateTemp(os.TempDir(), "unexisting")
+				if err != nil {
+					return "", err
+				}
+				err = os.Remove(f.Name())
+				if err != nil {
+					return "", err
+				}
+				return f.Name(), nil
+			},
+		},
+		"parent directory does not exist": {
+			expectedErr:        os.ErrNotExist,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				return "/random-path/which/does-not/exist", nil
+			},
+		},
+	}
+
+	hasOpenAt2Support := isOpenAt2Supported()
+	if !hasOpenAt2Support {
+		t.Logf("openat2 is not supported, will use mock function openAt2Mock")
+	} else {
+		t.Log("openat2 is supported, will use openat2 syscall.")
+	}
```

Comment:
instead of the conditional below on lines 664-668, you can define the openat2 func here:

```go
type openat2fndef func(int, string, *unix.OpenHow) (fd int, err error)
var openat2fn openat2fndef = nil
if !isOpenAt2Supported() {
    openat2fn = openAt2Mock
}
```